test/router.spec.js

```diff
@@ -450,6 +450,56 @@ describe('Restify Router', function () {
 
   });
 
+  describe('Nested routers via .add should process middlewares in order', function () {
+    it('Should process middlewares in order', function (done) {
+      var v1 = new Router();
+      var auth = new Router();
+      var register = new Router();
+
+      var first = function (req, res, next) {
+        req.test = [1];
```

Comment:
Good catch. Totally missed the reassignment 👍 

I have pushed the modified test case. Should work now.